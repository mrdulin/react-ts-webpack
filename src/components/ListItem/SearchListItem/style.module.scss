//导入公用的scss文件
//https://github.com/webpack-contrib/sass-loader#imports
@import '~styles/mixins';
//或者通过sass-loader的includePaths的option指定根路径
@import 'variables';

.list-item {
  display: flex;
  padding: 10px;
  background-color: $bg-color;
}

.col-left {
  width: 80px;
  margin-right: 10px;
}

.col-right {
  flex: 1;
}

.image {
  width: 100%;
  max-width: 100%;
}

.title {
  margin: 0;
  font-size: 16px;
  font-weight: normal;
}

.description {
  font-size: 14px;

  @include multi-ellipsis(2);
}
