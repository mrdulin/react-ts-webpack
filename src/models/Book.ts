/**
 * Created by dulin on 2017/6/29.
 */
import {processObjectPropName} from 'utils';
import {ISearchBook} from 'interfaces/book';

class Book {
  public id: number = -1;
  public title: string = '';
  public subtitle: string = '';
  public description: string = '';
  public image: string = '';
  public isbn: string = '';

  public constructor(book?: ISearchBook) {

    if (book) {
      const bookWithLowerCaseKey: any = processObjectPropName<ISearchBook>(book);

      Object.assign(this, bookWithLowerCaseKey);
    }

  }
  

  public isValid(): boolean {
    return this.id > 0;
  }
  
}


export default Book;
