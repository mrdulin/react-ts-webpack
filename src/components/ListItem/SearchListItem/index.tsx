import * as React from 'react';
import ListItem, {IListItemProps} from '../';
import Book from 'models/Book';

import * as style from './style.module.scss';

class SearchListItem extends ListItem<Book> {
  public static defaultProps: IListItemProps<Book> = {
    item: new Book(),
    onClick: (): void => {}
  };
  
  public render(): React.ReactElement<IListItemProps<Book>> {
    const {item}: {item: Book} = this.props;
    return (
      <div className={style.listItem} onClick={this.props.onClick}>
        <div className={style.colLeft}>
          <img className={style.image} src={item.image} alt="book-image"/>
        </div>
        <div className={style.colRight}>
          <h2 className={style.title}>{item.title}</h2>
          <p className={style.description}>{item.description}</p>          
        </div>
      </div>
    );
  }
}

export default SearchListItem;
