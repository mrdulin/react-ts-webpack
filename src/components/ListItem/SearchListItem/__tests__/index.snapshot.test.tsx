import * as React from 'react';
import * as renderer from 'react-test-renderer';

import SearchListItem from '../';
import Book from 'models/Book';

describe('component - SearchListItem snapshots test suites', () => {

  it('renders correctly', () => {

    const onClick = () => {};
    const book: Book = new Book({
      ID: 1,
      Title: 'react',
      SubTitle: '',
      Description: 'react is awesome',
      Image: '',
      isbn: ''
    });

    const component: renderer.Renderer = renderer.create(
      <SearchListItem item={book} onClick={onClick}/>
    );

    expect(component.toJSON()).toMatchSnapshot();
  });

});
