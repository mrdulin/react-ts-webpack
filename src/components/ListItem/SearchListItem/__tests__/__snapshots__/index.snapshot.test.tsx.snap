// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component - SearchListItem snapshots test suites renders correctly 1`] = `
<div
  className="listItem"
  onClick={[Function]}
>
  <div
    className="colLeft"
  >
    <img
      alt="book-image"
      className="image"
      src=""
    />
  </div>
  <div
    className="colRight"
  >
    <h2
      className="title"
    >
      react
    </h2>
    <p
      className="description"
    >
      react is awesome
    </p>
  </div>
</div>
`;
